packages/test/test-end-to-end-tests/src/test/SummarizerWithDataStoreRealization.spec.ts

```diff
@@ -0,0 +1,278 @@
+        assert(matrixHandle !== undefined, "SharedMatrix not found");
+        this.matrix = await matrixHandle.get();
+
+        this.matrix.insertRows(0, 3);
+        this.matrix.insertCols(0, 3);
+    }
+}
+const dataStoreFactory1 = new DataObjectFactory(
+    TestDataObjectType1,
+    TestDataObject1,
+    [SharedMap.getFactory(), SharedMatrix.getFactory()],
+    [],
+    [],
+    createDataStoreRuntime(),
+);
+const dataStoreFactory2 = new DataObjectFactory(
+    TestDataObjectType2,
+    TestDataObject2,
+    [SharedMap.getFactory(), SharedMatrix.getFactory()],
+    [],
+    [],
+    createDataStoreRuntime(),
+);
+const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+    runtime.IFluidHandleContext.resolveHandle(request);
+
+const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+    [dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+    [dataStoreFactory2.type, Promise.resolve(dataStoreFactory2)],
+    ],
+);
+const containerRuntimeFactoryWithDefaultDataStore =
+    getContainerRuntimeApi(pkgVersion).ContainerRuntimeFactoryWithDefaultDataStore;
+const runtimeFactory = new containerRuntimeFactoryWithDefaultDataStore(
+    dataStoreFactory1,
+    registryStoreEntries,
+    undefined,
+    [innerRequestHandler],
+    runtimeOptions,
+);
+
+async function createSummarizer(
+    provider: ITestObjectProvider,
+    container: IContainer,
+    summaryVersion?: string,
+): Promise<ISummarizer> {
+    return createSummarizerFromFactory(
+        provider,
+        container,
+        dataStoreFactory1,
+        summaryVersion,
+        containerRuntimeFactoryWithDefaultDataStore,
+        registryStoreEntries,
+    );
+}
+
+function createDataStoreRuntime(factory: typeof FluidDataStoreRuntime = FluidDataStoreRuntime) {
+    return mixinSummaryHandler(async (runtime: FluidDataStoreRuntime) => {
+        const obj: Provider<SearchContent> = (await DataObject.getDataObject(runtime)) as Provider<SearchContent>;
+        const searchObj = obj.SearchContent;
+        if (searchObj === undefined) {
+            return undefined;
+        }
+        // ODSP parser requires every search blob end with a line-feed character.
+        const searchContent = await searchObj.getSearchContent();
+        if (searchContent === undefined) {
+            return undefined;
+        }
+        const content = searchContent.endsWith("\n") ? searchContent : `${searchContent}\n`;
+        return {
+            // This is the path in snapshot that ODSP expects search blob (in plain text) to be for components
+            // that want to provide search content.
+            path: ["_search", "01"],
+            content,
+        };
+    }, factory);
+}
+
+/**
+ * Validates whether or not a GC Tree Summary Handle should be written to the summary.
+ */
+describeNoCompat("Summary with Search Blobs and DataStore realization during Summarization",
```

Comment:
nit: Maybe this should say "Summary where data store is loaded out of order". Same for the file name: "outOfOrderDataStoreRealization"?

Reply:
How about: SummarizerWithOutOfOrderDataStoreRealization.spec.ts ?

Reply:
that sounds good too :)